### DISCUSSION TITLE
Issue SH1107

### DISCUSSION THREAD

**Discussion by evanderkuy** [davetcc/tcMenu#466] [General]
Hi Dave,

(created a new ticket for this issue).

I am using the Waveshare OLED 128x128 (black/white) with a SH1107 driver and Adafruit libraries (because of my SW SPI pins as U8G2 does not support it for the SH1107.

In the file tcMenuAdaFruitGfx.h I have 2 errors in the functions drawCookieCutBitmap and drawCookieCutBitmap2bpp stating "incomplete type not allowed". I declare the gfx object with Adafruit_SH1107 gfx(128, 128, 18, 17, 2, -1, 13); 

I use Platformio on Ubuntu and I use these libraries :
	davetcc/tcMenu@^4.2.1
	davetcc/TaskManagerIO@^1.4.1
	davetcc/IoAbstraction@^4.0.2
	davetcc/tcUnicodeHelper@^1.0.0
	davetcc/SimpleCollections@^1.2.3
	adafruit/Adafruit GFX Library@^1.11.9
	adafruit/Adafruit SH110X@^2.1.10

I do not have the issue when I use a TFT display with a ST7789 driver as it runs perfectly fine using the same pins. When I create a project with only a Adafruit example it works fine.

Any clues, I have ordered 25 displays at Waveshare, so I hope I can still get it to work ;-).

**Comment by davetcc** [comment]
It is supported by U8G2 https://github.com/olikraus/u8g2/issues/501 so you could use it. There may not be a quick start for it, but it should be possible using the manual U8G2 plugin. The only difference is that you need to declare the variable yourself in your code and tell the plugin the variable name and type.

**Comment by evanderkuy** [comment]
Hi Dave,

Yes, I figured it out and noticed on the U8G2 Git repository they support the board with SW SPI. So now I have everything working. Speed is also much better and acceptable. Thank you so much for letting me know!
